### ISSUE TITLE
Fix the H5py Conda / Mantid Dependency Issue

### ISSUE THREAD

**Issue by glass-ships** [neutrons/drtsans#722]
Description
=====================================================================================
This task is to implement the solution to the problem described in task: #676 

## Short-term solution
TL;DR, the short-term solution described in #676 works :fireworks: .  
* To reproduce the results, do the following steps
  * Install `mantid` linking against system library.
    - disable conda env
    - enable devtools-7
    - build and install `mantid`
  * Clone `sans_backend` to local and install corresponding dependencies as usual
  * Install a customized version of `h5py` that links against system libhdf5
    - first, uninstall the `h5py` that was installed in previous step using `pip uninstall h5py`
    - now, do a install from source with
      ```bash
      HDF5_DIR=/usr/lib64 pip install --no-binary=h5py h5py
      ```
  * Install `drtsans` if needed with `pip install -e .`

* Several things to keep in mind:
  - this solution is mostly intended for the analysis cluster with RHEL_7, and it has not been tested with other os (although in theory it should also work)
  - any environment-wide update (like `conda update --all`) could break it again, therefore it is a short-term solution.

## Long-term solution
TL;DR the long-term solution requires extra time and efforts, and should be dealt with in a separate thread.
* The current deployment of `drtsans` uses `mantidframework` from [conda](https://anaconda.org/mantid/mantid-framework).
  - Due to reason unclear at the moment, the automatic building and publishing of `mantidframework` is broken at the moment.
  - Using the older version of `Mantidframework` from conda haven't raised any issue yet.
  - Fixing the building script for `Mantidframework` work is something beyond the scope of this task, and it is better handled in a separate story/defect/etc.

List of Tasks:
---------------
* [x]  Understand Story: Talk to owner
* [x]  Address Necessary Revisions to the Story
       
* [x]  Capture Significant Design Decisions
* [x]  Gather All Required Test Data
       
       
* [X]  Write & Run Unit Tests
* [x]  Implement Story (Update unit test as needed)
* [x]  Write & Run Integration Tests
* [x]  Submit Pull Request
* [ ]  Report Completion During Status Meeting

**Comment by glass-ships** [comment]
This issue was migrated from [code.ornl.gov](https://code.ornl.gov/sns-hfir-scse/sans/sans-backend/-/issues/677).
